src/In.ProjectEKA.HipService/DataFlow/DataFlowClient.cs

```diff
@@ -5,41 +5,61 @@ namespace In.ProjectEKA.HipService.DataFlow
     using System.Net.Http;
     using System.Text;
     using System.Threading.Tasks;
+    using CryptoHelper;
     using HipLibrary.Patient;
     using Hl7.Fhir.Serialization;
     using Logger;
     using Newtonsoft.Json;
     using Newtonsoft.Json.Serialization;
+    using Org.BouncyCastle.Crypto;
 
     public class DataFlowClient
     {
         private readonly ICollect collect;
         private readonly HttpClient httpClient;
+        private readonly ICryptoHelper cryptoHelper;
 
-        public DataFlowClient(ICollect collect, HttpClient httpClient)
+        public DataFlowClient(ICollect collect, HttpClient httpClient, ICryptoHelper cryptoHelper)
         {
             this.collect = collect;
             this.httpClient = httpClient;
+            this.cryptoHelper = cryptoHelper;
         }
 
         public async Task HandleMessagingQueueResult(HipLibrary.Patient.Model.DataRequest dataRequest)
         {
+            var senderKeyPair = cryptoHelper.GenerateKeyPair(dataRequest.KeyMaterial.Curve,
+                dataRequest.KeyMaterial.CryptoAlg);
+            var randomKeySender = cryptoHelper.GenerateRandomKey();
             (await collect.CollectData(dataRequest))
                 .Map(async entries =>
                 {
                     var serializer = new FhirJsonSerializer(new SerializerSettings());
                     var healthRecordEntries = entries.Bundles
                         .Select(bundle => new Entry(
-                            serializer.SerializeToString(bundle),
+                             EncryptData(dataRequest.KeyMaterial, serializer.SerializeToString(bundle), 
+                                 senderKeyPair, randomKeySender),
                             "application/json",
                             "MD5"))
                         .ToList();
-                    await SendDataToHiu(new DataResponse(dataRequest.TransactionId, healthRecordEntries),
+                    var keyMaterial = new KeyMaterial(dataRequest.KeyMaterial.CryptoAlg, dataRequest.KeyMaterial.Curve,
+                        new KeyStructure("", "", cryptoHelper.GetPublicKey(senderKeyPair)),
+                        randomKeySender);                    
+                    await SendDataToHiu(new DataResponse(dataRequest.TransactionId, healthRecordEntries, keyMaterial),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5073772)